Assets/Scripts/UI/Inventory/InventoryDragAndDrop.cs

```diff
@@ -83,8 +87,29 @@ private void MoveDraggedItemToSlot(InventorySlotUI moveTo)
         /// </summary>
         public void OnClickStart(InventorySlotUI slotBelowMouse)
         {
+            //Check if there is a slot, if not toss the item
             if (slotBelowMouse == null)
             {
+                if (_beingDragged.ItemStack.amount > 0)
+                {
+                    GameObject droppedItem = _beingDragged.ItemStack.itemInfo.ItemPrefab;
+
+                    //Spawn item in front of the player
+                    Vector3 playerPos = GameObject.FindWithTag("Player").transform.position;
+                    Vector3 playerFwd = GameObject.FindWithTag("Player").transform.forward;
+                    Vector3 updatedPos = new Vector3(playerPos.x, playerPos.y, playerPos.z) + playerFwd * 2f;
+                    UnityEngine.Object.Instantiate(droppedItem, updatedPos, GameObject.FindWithTag("Player").transform.rotation);
+
+                    //Decrement Item
+                    _beingDragged.ItemStack.amount -= 1;
+                }
+                if (_beingDragged.ItemStack.amount <= 0)
+                {
+                    //Hide UI Drag Element
+                    _beingDragged.ShowItem(null);
+                    _beingDragged.ItemStack.amount = 0;
```

Comment:
This might cause bugs. It should probably remove the item stack rather than hiding a stack with a count of 0.

Item stacks are stored in at least 2 places: InventoryUIBase._itemsInSlotIndexes (a dictionary), and InventoryBase._items (a hashset). Make sure whatever stack is removed gets removed from both. Or whatever works, e.g. it might be better to change InventoryBase to use the dictionary's keys rather than needing to update a hashset to match.